orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v1schema/graph/transform/ConfigStageInjectionTransform.java

```diff
@@ -97,7 +103,7 @@ private static void dfs(String stageId,
 
     for (String n : stage.getRequisiteStageRefIds()) {
       Status status = state.get(n);
-      if (status == Status.VISITING) throw new IllegalStateException("Cycle detected in graph");
+      if (status == Status.VISITING) throw new IllegalTemplateConfigurationException("Cycle detected in graph (discovered on stage: " + stageId + ")");
```

Comment:
Everything else in here uses `String.format()` instead of string concatenation, would be nice to be consistent.